packages/core/src/common/audioAlertManager.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  Annotations,
+  breachedThreshold,
+  DataStream as SynchroChartsDataStream,
+  getThresholds,
+  MinimalViewPortConfig,
+} from '@synchro-charts/core';
+import { DataStream } from '../data-module/types';
+import { AudioAlert } from './types';
+import { leastSevere, liveDataTimeBufferMs } from './constants';
+import { getVisibleData } from './dataFilters';
+import { getDataPoints } from './getDataPoints';
+import { Threshold } from './types';
+import { viewportEndDate } from './viewport';
+import { AudioAlertPlayer, calculateSeverity } from './audioAlertPlayer';
+
+export const audioAlertPlayer = new AudioAlertPlayer();
+
+// Returns true if viewport is in Live Mode or if viewport.end is past the current time and date
+export const isLiveData = (viewport: MinimalViewPortConfig): boolean => {
+  return viewportEndDate(viewport).getTime() + liveDataTimeBufferMs > Date.now();
+};
+
+// Plays audio alert if in live mode and newly pushed point breaches a threshold
+export const playThresholdAudioAlert = ({
+  dataStreams,
+  viewport,
+  annotations,
+}: {
+  dataStreams: DataStream[];
+  viewport: MinimalViewPortConfig;
+  annotations: Annotations | undefined;
+}): Map<string, AudioAlert> | undefined => {
+  if (!isLiveData(viewport)) {
+    return;
+  }
+  const thresholds = getThresholds(annotations);
+  if (thresholds.length === 0) {
+    return;
+  }
+
+  // Audio alerts are only applied to live data
```

Comment:
i'm confused as to how this only applies to live data? it looks like it applies to any data within the current viewport

Reply:
I should move this comment to earlier in the code where I call on `isLiveData`.